.github/workflows/static_analysis.yml

```diff
@@ -0,0 +1,52 @@
+# contains tests that can run without compiling LightGBM, like
+# linters and documentation checks.
```

Comment:
Maybe move R lint check here for better logical grouping? Does it require all our R stuff installation routine or we can isolate this check in Docker like ASAN/UBSAN?
https://github.com/microsoft/LightGBM/blob/fc1b930026a2a376d8d9174c2c7662b7185885b5/.github/workflows/r_package.yml#L44-L47

Reply:
`r-package-check-docs` does require all of our stuff, unfortunately. The library has to be installed (https://github.com/microsoft/LightGBM/blob/master/.ci/test_r_package.sh#L109) before you can re-build the docs with `roxygen2::roxygenize()`

Reply:
OK, I see. But seems that other R jobs also require building but are run inside a Docker. For instance, 
https://github.com/microsoft/LightGBM/blob/98e6f719c5111a8bc631877c3d3d9c55e360ae5d/.github/workflows/r_package.yml#L173-L180
or 
https://github.com/microsoft/LightGBM/blob/98e6f719c5111a8bc631877c3d3d9c55e360ae5d/.github/workflows/r_valgrind.yml#L23-L25

I wonder whether we can do the same for docs? I think the only thing we lose is finer control of used R version and CRAN/CMake type of installation. But is it important for docs check?

Reply:
oh true, that's a good point! Ok let me try that, I like that idea.

Reply:
ok yeah that worked! That job only takes 7 minutes, I'm very happy with that and very comfortable grouping that one with `check-docs` and `lint`.